### ISSUE TITLE
Rspec test is passing - but the source code isan

### ISSUE THREAD

**Issue by elsstrange** [makersacademy/common-knowledge#251]
I have an rspec feature test for adding a new bookmark in the Bookmarks challenge.

I've just added a line `Bookmark.create` into my app.rb controller.  This method doesn't yet exist in my Bookmark class.

I would expect the test to fail, surfacing an error about calling a method that doesn't exist.

Instead the test passes. 

I put some statements to log to the terminal the routes that the feature test passes through, and I can see that it gets stuck at the point where it calls Bookmark.create - but it's still passing!

I suspect that I must have configured something oddly in Rspec, but can't figure out a good search term to find other people with the issue.

**Comment by elsstrange** [comment]
Further investigations:

If I implement a .create method on Bookmark, my test starts passing and failing as normal - 

I can change one of my expectations to differ from my set up and get a failure.
I can hard code a value into the Bookmark.create method call and get a failure.

So it's like, when it runs into trouble, my RSpec test just stops running, claims to have passed, and then moves onto the next test.

**Comment by dearshrewdwit** [comment]
1. What are the requirements for each expectation you have in the feature test to pass?

2. If the requirements are met to pass the test, how can you get visibility on them during the test run?

**Comment by elsstrange** [comment]
Did some more digging (thanks @amittauro) and reached this conclusion:

The feature test was:
```ruby
expect(page).to have_content "IMDB"
expect(page).to have_content "http://www.imdb.com"
expect(current_path).to eq "/bookmarks"
```

We added into the test
```
p page.body
```
And we realised that it was loading the Sinatra error page, explaining that .create method doesn't exist. Viewing the Sinatra error page in a browser, it turned out that the Sinatra error page lists the content of the POST request, meaning that the content expected by the tests was present on the page.

Possible ways around this:
1. Be more specific in the content that your test expects, e.g. `expect(page).to have_content "IMDB - http://www.imdb.com"`
2. Include an `expect(page).not_to have_content` test that looks for some property of the error page
3. Switch off the Sinatra setting that shows error pages (though not sure what would show instead)

What this also highlights is that because the feature test is driving a browser, it is unaware of errors in the Ruby source code.  Where a unit test, which interacts directly with the source code, hits an error and logs it to the terminal, Capybara's feature testing will just continue until it can no longer follow the Capybara instructions.
